### PR TITLE
Add feature page to new UI and refine search feature

### PR DESCRIPTION
**Pull request by jmezach**
We've added the feature page to the new Blazor UI and refined the search feature so that it works a little bit better.

### CODE REVIEW

File: src/Augurk.UI/Components/ComponentReplacer.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+using System;
```
Comment (jmezach) [inline src/Augurk.UI/Components/ComponentReplacer.cs:1]: Missing license

File: src/Augurk.UI/Components/ArgumentDisplay.razor.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.argument {
```
Comment (jmezach) [inline src/Augurk.UI/Components/ArgumentDisplay.razor.css:1]: Missing license

File: src/Augurk.UI/Components/KeywordDisplay.razor.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.keyword {
```
Comment (jmezach) [inline src/Augurk.UI/Components/KeywordDisplay.razor.css:1]: Missing license

File: src/Augurk.UI/Components/Markdown.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+using Markdig;
```
Comment (jmezach) [inline src/Augurk.UI/Components/Markdown.cs:1]: Missing license

File: src/Augurk.UI/Components/SequenceDisplay.razor [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+-->
+@using System.Web
+@inject IJSRuntime jsRuntime
+
+<div @ref=Div class="sequence">
+
+</div>
+@if(!String.IsNullOrWhiteSpace(Label))
+{
+<span class="label">@Label</span>
```
Comment (jmezach) [inline src/Augurk.UI/Components/SequenceDisplay.razor:24]: Indentation?

File: src/Augurk.UI/Components/SequenceDisplay.razor.css [hunk 6]
```diff
@@ -0,0 +1,16 @@
+div.sequence{
```
Comment (jmezach) [inline src/Augurk.UI/Components/SequenceDisplay.razor.css:1]: Missing license

File: src/Augurk.UI/Components/StepDisplay.razor.css [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.step {
```
Comment (jmezach) [inline src/Augurk.UI/Components/StepDisplay.razor.css:1]: Missing license

File: src/Augurk.UI/Components/StepTable.razor.css [hunk 8]
```diff
@@ -0,0 +1,3 @@
+.argument {
```
Comment (jmezach) [inline src/Augurk.UI/Components/StepTable.razor.css:1]: Missing license

File: src/Augurk.UI/Components/UmlDisplay.razor [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+-->
+@using System.Web
+@inject IJSRuntime jsRuntime
+
+<div class="uml">
+    <canvas @ref=Canvas></canvas>
+</div>
+@if(!String.IsNullOrWhiteSpace(Label))
+{
+<span class="label">@Label</span>
```
Comment (jmezach) [inline src/Augurk.UI/Components/UmlDisplay.razor:24]: Indentation?

File: src/Augurk.UI/Components/UmlDisplay.razor.css [hunk 10]
```diff
@@ -0,0 +1,11 @@
+div.uml{
```
Comment (jmezach) [inline src/Augurk.UI/Components/UmlDisplay.razor.css:1]: Missing license

File: src/Augurk.UI/Pages/FeaturePage.razor [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+            @foreach (var scenario in @CurrentFeature?.Scenarios)
+            {
+                <ScenarioDisplay Scenario="@scenario" />
+            }
+        </Row>
+    }
+</div>
+
+@code {
+
```
Comment (jmezach) [inline src/Augurk.UI/Pages/FeaturePage.razor:57]: Unnecessary blank line

File: src/Augurk.UI/Pages/FeaturePage.razor.css [hunk 12]
```diff
@@ -0,0 +1,3 @@
+.description {
```
Comment (jmezach) [inline src/Augurk.UI/Pages/FeaturePage.razor.css:1]: Missing license

File: src/Augurk.UI/Pages/Index.razor [hunk 13]
```diff
@@ -21,6 +21,15 @@
 <div class="container-fluid">
     <div class="row">
         <Searchbox />
+        @if (_searching)
+        {
+        <div class="col-md-6 offset-md-3" style="margin-top: 5px;">
```
Comment (jmezach) [inline src/Augurk.UI/Pages/Index.razor:26]: Identation?

File: src/Augurk.UI/Program.cs [hunk 14]
```diff
@@ -16,6 +16,7 @@
 using System;
```
Comment (jmezach) [inline src/Augurk.UI/Program.cs:16]: Missing license

File: src/Augurk.UI/Services/SearchService.cs [hunk 15]
```diff
@@ -27,27 +27,36 @@ public class SearchService {
         get; private set;
     }
 
+    public bool Searching {
```
Comment (jmezach) [inline src/Augurk.UI/Services/SearchService.cs:30]: Open brace on newline

File: src/Augurk.UI/_Imports.razor [hunk 16]
```diff
@@ -7,4 +7,8 @@
 @using Microsoft.AspNetCore.Components.WebAssembly.Http
 @using Microsoft.JSInterop
 @using Augurk.UI
+@using Augurk.UI.Components
```
Comment (jmezach) [inline src/Augurk.UI/_Imports.razor:10]: Missing license

File: src/Augurk.UI/Components/FeatureResult.razor [hunk 17]
```diff
@@ -10,10 +10,11 @@
  Unless required by applicable law or agreed to in writing, software
```
Comment (marktaling) [inline src/Augurk.UI/Components/FeatureResult.razor:10]: Update license to include 2021

File: src/Augurk.UI/Components/Searchbox.razor [hunk 18]
```diff
@@ -10,15 +10,16 @@
  Unless required by applicable law or agreed to in writing, software
```
Comment (marktaling) [inline src/Augurk.UI/Components/Searchbox.razor:10]: Update license to include 2021

File: src/Augurk.UI/Components/SequenceDisplay.razor (hunk 5 above)
Comment (marktaling) [reply src/Augurk.UI/Components/SequenceDisplay.razor:24]: Might have been a bit hasty once it finally worked :P

File: src/Augurk.UI/Components/TagDisplay.razor [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+@if (Tags.Count() > 0)
+{
+    <Row>
+        <p>
+            @if (Tags.Count() == 1) { <i class="bi-tag-fill">&nbsp;</i> }
```
Comment (marktaling) [inline src/Augurk.UI/Components/TagDisplay.razor:21]: Perhaps not inline the html for readability?

File: src/Augurk/Managers/FeatureManager.cs [hunk 20]
```diff
@@ -307,24 +307,40 @@ public async Task<DbFeature> InsertOrUpdateFeatureAsync(Feature feature, string
...
+                {
                     // Create a new feature
                     var processor = new FeatureProcessor();
                     string parentTitle = processor.DetermineParent(feature);
-                    dbFeature = new DbFeature(feature, productName, groupName, parentTitle, version);
+                    dbFeature = new DbFeature(feature, productName, groupName, parentTitle, version) { Hash = hash };
                     await session.StoreAsync(dbFeature, dbFeature.GetIdentifier());
                 }
 
+                var wronglyStampedFeatures = session.Query<DbFeature, Features_ByTitleProductAndGroup>()
```
Comment (marktaling) [inline src/Augurk/Managers/FeatureManager.cs:329]: Some comment for explaination might not be amiss

**Review by marktaling — COMMENTED** [review]
We've got some cleaning up to do

File: src/Augurk.Entities/DisplayableFeature.cs [hunk 21]
```diff
@@ -1,12 +1,12 @@
 ﻿/*
  Copyright 2014-2015, Mark Taling
```
[the remark this answers is not in the document]
Comment (jmezach) [reply src/Augurk.Entities/DisplayableFeature.cs:2]: Fixed :)

File: src/Augurk.Entities/Feature.cs [hunk 22]
```diff
@@ -66,6 +66,7 @@ public string CalculateHash()
             var sb = new StringBuilder();
             sb.AppendLine(String.Join(" ", Tags));
             sb.AppendLine(Title);
+            sb.AppendLine(Description);
             if(Background != null){
                 sb.AppendLine(Background.ToString());
```
[the remark this answers is not in the document]
Comment (jmezach) [reply src/Augurk.Entities/Feature.cs:71]: Fixed

File: src/Augurk.Entities/Scenario.cs [hunk 23]
```diff
@@ -63,6 +63,7 @@ public override string ToString()
             }
 
             var sb = new StringBuilder();
+            sb.AppendLine(Description);
             foreach(var step in Steps)
```
[the remark this answers is not in the document]
Comment (jmezach) [reply src/Augurk.Entities/Scenario.cs:67]: Fixed

File: src/Augurk.UI/Components/ComponentReplacer.cs (hunk 1 above)
Comment (jmezach) [reply src/Augurk.UI/Components/ComponentReplacer.cs:1]: Fixed

File: src/Augurk.UI/Components/Markdown.cs (hunk 4 above)
Comment (jmezach) [reply src/Augurk.UI/Components/Markdown.cs:1]: Fixed

File: src/Augurk.UI/Program.cs (hunk 14 above)
Comment (jmezach) [reply src/Augurk.UI/Program.cs:16]: Fixed

File: src/Augurk.UI/Services/SearchService.cs (hunk 15 above)
Comment (jmezach) [reply src/Augurk.UI/Services/SearchService.cs:30]: Fixed

File: src/Augurk/Managers/FeatureManager.cs (hunk 20 above)
Comment (jmezach) [reply src/Augurk/Managers/FeatureManager.cs:329]: Fixed

File: src/Augurk.UI/_Imports.razor (hunk 16 above)
Comment (jmezach) [reply src/Augurk.UI/_Imports.razor:10]: Unfortunately it seems that the `file_header_template` we've added to the `.editorconfig` only applies to C# and VB files, but not to Razor, CSS, or other files. So we'll have to do these manually, or remove them entirely I guess. @marktaling Thoughts?

Comment (marktaling) [reply src/Augurk.UI/_Imports.razor:10]: Half manual / half automatic doesn't sound all to great to me. There are some vscode extensions out there which also do this for other file types, perhaps we should take a look at those?

Comment (jmezach) [reply src/Augurk.UI/_Imports.razor:10]: I've added an extension to the workspace recommendations and configured it for Razor and CSS files.

File: src/Augurk.UI/Components/FeatureResult.razor (hunk 17 above)
Comment (jmezach) [reply src/Augurk.UI/Components/FeatureResult.razor:10]: As discussed we've simplified the license header to not include a year

**Pull request merged**